### PR TITLE
MBS-9038 & MBS-9039: Handle new patronage and crowdfunding URL relationships

### PR DESCRIPTION
**Pull request by y-van-z**
For ChangeTip, Flattr, Kickstarter, Indiegogo, Patreon, PayPal.Me and Tipeee:
- auto-detect (new) `patronage` or `crowdfunding` link type and clean URLs,
- display in the sidebar with specific favicon and label.

(Keep patronage pages grouped in the `/relationships` tab.)
(Keep  `crowdfunding` label in the `/relationships` tab.)

### CODE REVIEW

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -765,6 +772,26 @@ const CLEANUPS = {
...
       return url;
     }
+  },
+  patronage: {
+    match: [
+      new RegExp("^(https?://)?(www\\.)?changetip\\.com/tipme/[^/?#]", "i"),
+      new RegExp("^(https?://)?[^/?#]+\\.tip.me([/?#].*)?$", "i"),
+      new RegExp("^(https?://)?(www\\.)?flattr\\.com/profile/[^/?#]", "i"),
+      new RegExp("^(https?://)?(www\\.)?patreon\\.com/[^/?#]", "i"),
+      new RegExp("^(https?://)?(www\\.)?paypal\\.me/[^/?#]", "i")
```
Comment (mwiencek) [inline root/static/scripts/edit/URLCleanup.js]: Comma is missing there (I assume that's what's causing parse errors). You can safely add a comma at the end of every line, like in other languages.

Comment (y-van-z) [inline root/static/scripts/edit/URLCleanup.js]: Thank you, (no) trailing comma is mostly a style preference, good to know you accept both.

File: lib/MusicBrainz/Server/Entity/URL/ChangeTip.pm [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
```
Comment (uklauer) [inline lib/MusicBrainz/Server/Entity/URL/ChangeTip.pm]: The current address of the FSF is `51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA`.

However, it actually [recommends](https://www.gnu.org/licenses/gpl-howto.html) using the following text for the final paragraph nowadays:

    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.

Comment (y-van-z) [inline lib/MusicBrainz/Server/Entity/URL/ChangeTip.pm]: Thank you, now up-to-date.

**Comment by mwiencek** [comment]
Thanks 👍

**Comment by tchoulihan** [comment]
When would this change get pushed up to musicbrainz? I'd like to start linking artist donation buttons to my music site if/when this exists.

**Comment by mwiencek** [comment]
@tchoulihan It'll get pushed to the main site on Monday, or you can already try it through https://beta.musicbrainz.org (the beta site uses the same database as main one).

**Pull request merged**